catroid/src/test/java/org/catrobat/catroid/test/content/actions/SetThreadColorActionTest.kt

```diff
@@ -0,0 +1,128 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.actions
+
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.scenes.scene2d.Action
+import com.badlogic.gdx.scenes.scene2d.actions.Actions
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction
+import junit.framework.Assert.assertEquals
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.content.Scope
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.content.actions.SetThreadColorAction
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.test.StaticSingletonInitializer.Companion.initializeStaticSingletonMethods
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class SetThreadColorActionTest {
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        initializeStaticSingletonMethods()
+        sprite = Sprite("testSprite")
+        createTestableSetThreadColorAction(
+            sprite,
+            SequenceAction(), Formula(COLOR_TEST_VALUE_BLACK)
+        )?.act(1.0f)
+    }
+
+    @Test
+    fun testNormalBehaviorColorChange() {
+        assertEquals(Color.BLACK, sprite?.embroideryThreadColor)
+        createTestableSetThreadColorAction(
+            sprite, SequenceAction(),
+            Formula(COLOR_TEST_VALUE_WHITE)
+        )?.act(1.0f)
+        assertEquals(Color.WHITE, sprite?.embroideryThreadColor)
+        createTestableSetThreadColorAction(
+            sprite, SequenceAction(),
+            Formula(COLOR_TEST_VALUE_BLUE)
+        )?.act(1.0f)
+        assertEquals(Color.BLUE, sprite?.embroideryThreadColor)
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun testNullSprite() {
+        val action = sprite?.actionFactory?.createSetThreadColorAction(
+            null,
+            SequenceAction(), Formula(COLOR_TEST_VALUE_BLACK)
+        )
+        action?.act(1.0f)
+    }
+
+    @Test
+    fun testNullFormula() {
+        createTestableSetThreadColorAction(sprite, SequenceAction(), null)?.act(1.0f)
+        assertEquals(COLOR_INITIALIZED_VALUE, sprite?.embroideryThreadColor)
+    }
+
+    @Test(expected = StringIndexOutOfBoundsException::class)
+    fun testTooShortFormula() {
+        createTestableSetThreadColorAction(
+            sprite,
+            SequenceAction(), Formula("#000")
+        )?.act(1.0f)
+    }
+
+    @Test(expected = NumberFormatException::class)
+    fun testInvalidColorCode() {
+        createTestableSetThreadColorAction(
+            sprite,
+            SequenceAction(), Formula(TEST_INVALID_VALUE)
+        )?.act(1.0f)
+        assertEquals(COLOR_INITIALIZED_VALUE, sprite?.embroideryThreadColor)
+    }
+
+    private fun createTestableSetThreadColorAction(sprite: Sprite?, sequence: SequenceAction?, color: Formula?): Action? {
```

Comment:
If `android.graphics.Color.argb` can be substituted then we will not need this method anymore.